aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -46,7 +49,186 @@ bool should_use_th_sort(const Tensor &self, int64_t dim) {
 
 std::vector<int64_t> infer_dense_strides_dim_last(const Tensor & self, int64_t dim);
 
-// We perform a vectorized segmented sort in cub.
+void fillSliceWithIndex(Tensor& t,
+                        int dim) {
+  if (t.numel()) {
+    auto sizes = DimVector(t.dim(), 1);
+    sizes[dim] = t.sizes()[dim];
+    int64_t rangelen = 1;
+    rangelen = std::max(t.sizes()[dim], rangelen);
+    auto range = at::arange(rangelen, t.options());
+    auto rangeview = range.view(sizes);
+    t.copy_(rangeview);
+  }
+}
+
+// In alignment with default sort on a c++ map, this function
+// will permute key and value tensors identically, and
+// in such a way that the 'key' tensor is ordered numerically
+void sortKeyValueInplace(Tensor& key,
+                         Tensor& value,
+                         int dim, bool dir) {
+  TORCH_CHECK(key.sizes() == value.sizes(),
+              "Key tensor must have same size as value tensor");
+  int dims = value.dim();
+  dims = dims == 0 ? 1 : dims;
```

Comment:
See https://github.com/pytorch/pytorch/pull/55392/files#r619534695 and https://github.com/pytorch/pytorch/pull/55392/files#r619541581 remove this logic here